### PR TITLE
[WIP] Returning to the latest job page from annotation view

### LINKED ISSUE #3101: Finishing the job should take you to the most recent pagination page
### My actions before raising this issue
- [ x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [ x] Searched [past issues](/issues)

### Expected Behaviour

When pressing 'Finish the job', we are returned to the job page in the task of the job we just finished or to proceed to the next job in the task (latter preferrable). This is particularly important when there are 100s of jobs per task.

### Current Behaviour

When pressing 'Finish the job', we are returned to the first job page in the task. 

### Your Environment

- Git hash commit (`git log -1`):
- Docker version `docker version` (e.g. Docker 17.0.05):
- Are you using Docker Swarm or Kubernetes?
- Operating System and version (e.g. Linux, Windows, MacOS):
- Code example or link to GitHub repo or gist to reproduce problem:
- Other diagnostic information / logs:
    <details>
    <summary>Logs from `cvat` container</summary>
    </details>

### Next steps
You may [join our Gitter](https://gitter.im/opencv-cvat/public) channel for community support.

### PR DESCRIPTION
**Pull request by dreaquil**
### Motivation and context
Resolved #3101 

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@dreaquil , what is a motivation behind the PR? What is it going to fix?

**Comment by bsekachev** [comment]
@nmanovic 

Issue #3101

@dreaquil  sorry for the long response. I was on vacation, I'll take a look into the PR the nearest time.

File: cvat-ui/src/components/task-page/job-list.tsx [hunk 1]
```diff
@@ -13,6 +13,8 @@ import Button from 'antd/lib/button';
 import Text from 'antd/lib/typography/Text';
 import moment from 'moment';
 import copy from 'copy-to-clipboard';
+import { CombinedState } from 'reducers/interfaces';
+import { connect } from 'react-redux';
```
Comment (bsekachev) [inline cvat-ui/src/components/task-page/job-list.tsx:17]: All 3rdparty import should occur before internal imports

File: cvat-ui/src/reducers/tasks-reducer.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright (C) 2020 Intel Corporation
+// Copyright (C) 2021 Intel Corporation
```
Comment (bsekachev) [inline cvat-ui/src/reducers/tasks-reducer.ts:1]: The right option is 2020-2021.
We should keep initial year also according to internal rules.

File: cvat-ui/src/components/task-page/job-list.tsx [hunk 3]
```diff
@@ -257,6 +274,12 @@ function JobListComponent(props: Props & RouteComponentProps): JSX.Element {
         },
     ];
 
+    const onChange = function (newPage: number): void {
+        // console.log("props.currentJobsPAge:", currentJobsPage);
+        // console.log("update page to:", newPage);
```
Comment (bsekachev) [inline cvat-ui/src/components/task-page/job-list.tsx:279]: These console.log should be removed from production code

File: cvat-ui/src/components/task-page/job-list.tsx [hunk 4]
```diff
@@ -330,9 +353,16 @@ function JobListComponent(props: Props & RouteComponentProps): JSX.Element {
                 columns={columns}
                 dataSource={data}
                 size='small'
+                pagination={{
+                    current: currentJobsPage,
+                    onChange,
+                    position: ['topRight', 'bottomRight'],
+                    defaultPageSize: 10,
+                    showSizeChanger: true,
```
Comment (bsekachev) [inline cvat-ui/src/components/task-page/job-list.tsx:361]: Should we declare these parameters explicitly?

File: cvat-ui/src/components/task-page/job-list.tsx [hunk 5]
```diff
@@ -26,6 +28,19 @@ const baseURL = core.config.backendAPI.slice(0, -7);
...
+
+interface StateToProps {
+    currentJobsPage: number;
+}
+
+function mapStateToProps(state: CombinedState): StateToProps {
+    console.log('State2:', state);
+    return {
+        currentJobsPage: state.tasks.current[0].currentJobsPage,
+    };
```
Comment (bsekachev) [inline cvat-ui/src/components/task-page/job-list.tsx:43]: I would suggest just use hook useSelector inside the functional component below

https://react-redux.js.org/api/hooks#useselector

Comment (bsekachev) [reply cvat-ui/src/components/task-page/job-list.tsx:43]: And useDispatch to update redux state

https://react-redux.js.org/api/hooks#usedispatch

File: cvat-ui/src/reducers/tasks-reducer.ts [hunk 6]
```diff
@@ -72,6 +73,15 @@ export default (state: TasksState = defaultState, action: AnyAction): TasksState
                 gettingQuery: { ...action.payload.query },
             };
         }
+
+        case TasksActionTypes.UPDATE_CURRENT_JOBS_PAGE: {
+            return {
+                ...state,
+                current: state.current.map((current, i) =>
+                    (i === 0 ? { ...current, currentJobsPage: action.payload.pageNumber } : current)),
```
Comment (bsekachev) [inline cvat-ui/src/reducers/tasks-reducer.ts:81]: Probably ``i===0`` is not correct condition. We should use ``taskID`` from payload.

**Comment by bsekachev** [comment]
Hi, @dreaquil 

I looked at the patch
The common idea and approach looks nice
There are some comments in the code and also it does not work correctly with applied filters and sortings

**Comment by bsekachev** [comment]
Greetings, @dreaquil 

Just I want to know if you have any plans to continue working on the PR.

**Comment by dreaquil** [comment]
Hi @bsekachev,

Thanks for your feedback on the PR. I'd like to continue work on the PR however, it seems to be difficult to find time as there's been a priority shift. If there's anyone that would like to pick this up, they're free to do so. Otherwise, I'll try and find some time in the coming weeks.

**Comment by bsekachev** [comment]
I will close the PR for now. Do not hesitate to reopen it if you find time to implement the feature.

**Pull request closed without merging**